mesatee_services/fns/sgx_trusted_lib/src/global.rs

```diff
@@ -132,10 +137,10 @@ extern "C" fn c_save_file_for_all_participants(
                 out_file_id[..file_id_len].copy_from_slice(file_id.as_bytes());
                 file_id_len as c_int
             } else {
-                out_file_id_buf_size as c_int - file_id_len as c_int
+                FFI_BUFFER_NOT_ENOUGH_ERROR
```

Comment:
The current implementation of this branch implies that even if we successfully create and save the output file in the rust layer, the python script still gets  FFI_BUFFER_NOT_ENOUGH_ERROR, which will finally `raise mesatee_error(space, "Cannot save file")`.
I suggest add the buffer_size check before calling `save_file_for_all_participants`.  

Currently the type of a file_id is `String`, it might be a little weird to get the upper bound of its length to do the check. I think it's necessary to improve the internal data structure type definitions in the future.😄

Reply:
We don't know the `file_id_len` before calling `save_file_for_all_participants`. Right now, the max length of  is 36 chars, so we are okay now.